File: src/main/java/de/php_perfect/intellij/ddev/actions/DdevRestartAction.java
package de.php_perfect.intellij.ddev.actions;

import com.intellij.openapi.project.Project;
import de.php_perfect.intellij.ddev.cmd.Description;
import de.php_perfect.intellij.ddev.config.DdevConfigurationProvider;
import org.jetbrains.annotations.NotNull;

public class DdevRestartAction extends DdevRunAction {
    public DdevRestartAction() {
        super("restart");
    }

    protected boolean isActive(@NotNull Project project) {
        final DdevConfigurationProvider ddevConfigurationProvider = DdevConfigurationProvider.getInstance(project);

        if (!ddevConfigurationProvider.isInstalled()) {
            return false;
        }

        Description status = ddevConfigurationProvider.getStatus();

        if (status == null) {
            return false;
        }

        return status.getStatus() == Description.Status.RUNNING || status.getStatus() == Description.Status.UNDEFINED;
    }
}

File: src/main/java/de/php_perfect/intellij/ddev/actions/DdevRestartAction.kt
package de.php_perfect.intellij.ddev.actions

import com.intellij.openapi.project.Project
import de.php_perfect.intellij.ddev.cmd.DdevRunner
import de.php_perfect.intellij.ddev.cmd.Description
import de.php_perfect.intellij.ddev.state.DdevStateManager
import de.php_perfect.intellij.ddev.state.State

class DdevRestartAction : DdevRunAction() {
    override fun run(project: Project) {
        DdevRunner.getInstance().restart(project)
    }

    override fun isActive(project: Project): Boolean {
        val state: State = DdevStateManager.getInstance(project).state
        if (!state.isAvailable || !state.isConfigured) {
            return false
        }
        val description: Description? = state.description
        return description == null || description.status == Description.Status.RUNNING
    }
}
